fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AuthorizerStmtVisitor.java

```diff
@@ -260,23 +277,40 @@ public Void visitDeleteStatement(DeleteStmt statement, ConnectContext context) {
         }
         Map<TableName, Relation> allTouchedTables = AnalyzerUtils.collectAllTableAndViewRelations(statement);
         allTouchedTables.remove(statement.getTableName());
-        checkSelectTableAction(context, allTouchedTables);
+        try {
+            Map<TableName, Set<String>> tableColumns = AnalyzerUtils.collectAllSelectTableColumns(statement);
+            tableColumns.remove(statement.getTableName());
+            checkCanSelectFromColumns(context, tableColumns, allTouchedTables);
+        } catch (ErrorReportException e) {
```

Comment:
ErrorReportException is a top-level error handling mechanism. Generally, we do not catch ErrorReportException. CheckCanSelectFromColumns can directly throw AccessDeniedException.